app/Http/Controllers/BookingController.php

```diff
@@ -232,25 +368,72 @@ public function addTable(Request $request)
 
         if ($seconds < 0) {
             Session::flash('error', 'The specified date and time must be greater than or equal to the next 30mins from now.');
+            $output = [
+                'status'    => 'failure',
+                'message'   => 'The specified date and time must be greater than or equal to the next 30mins from now.',
+            ];
+            return json_encode($output);
         } else {
             //Convert to correct format
             $request->date = str_replace('/', '-', $request->date);
-            Cart::add([
-                'id' => time(),
-                'name' => $table->label,
-                'quantity' => 1,
-                'price' => round($table->cost, 2),
-                'attributes' => [
-                    'item_id' => $table->id,
-                    'date' => $request->date,
-                    'duration' => $request->duration,
-                    'type' => 'table',
-                ],
-            ]);
-            Session::flash('success', 'Table added to cart');
+
+            $cartData = $this->prepareCartData($request, $table);
+
+            $this->addToCart($cartData);
+
+            $this->clearBookingSession();
+
+            $output = [
+                'status'    => 'success',
+                'message'   => 'Table added to cart',
+            ];
+
+            $this->setTableToOnHold($table);
+
+            return json_encode($output);
         }
 
-        return redirect()->back();
+        // return redirect()->back();
+    }
+
+    private function prepareCartData($request ,$table)
```

Comment:
What was the essence of separating out this method? Looks like you had some ideas.